File: extractor-appender/appender-sql/src/main/java/ru/mydesignstudio/database/metadata/extractor/database/ExtractHelper.java
package ru.mydesignstudio.database.metadata.extractor.database;

import java.util.Collection;
import java.util.HashSet;
import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.BeanPropertyRowMapper;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Component;

@Component
public class ExtractHelper {
  @Autowired
  private JdbcTemplate jdbcTemplate;

  public <T> Collection<T> extract(@NonNull String query, @NonNull Object[] params, Class<T> targetClass) {
    final List<T> extractedItems = jdbcTemplate.query(query, params, new BeanPropertyRowMapper<>(targetClass));
    return new HashSet<>(extractedItems);
  }
}

File: extractor-source/source-core/src/main/kotlin/ru/mydesignstudio/database/metadata/extractor/extract/helper/ExtractHelper.kt
package ru.mydesignstudio.database.metadata.extractor.extract.helper

import org.springframework.jdbc.core.BeanPropertyRowMapper
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.jdbc.core.RowMapper
import java.util.*

@Deprecated("Useless class, it's getter get rid of JdbcTemplate and use PreparedStatement instead")
class ExtractHelper (private val jdbcTemplate: JdbcTemplate) {

    fun <T> extract(query: String, params: List<String>, targetClass: Class<T>): Collection<T> {
        var queryParams = params.toTypedArray();
        val extractedItems = jdbcTemplate.query(query, queryParams, BeanPropertyRowMapper(targetClass))
        return HashSet(extractedItems)
    }

    fun <T> extract(query: String, params: List<String>, mapper: RowMapper<T>): Collection<T> {
        val queryParams = params.toTypedArray()
        val extractedItems = jdbcTemplate.query(query, mapper, queryParams)
        return extractedItems
    }
}